From d4186c1e1dec284864ba41b0d7878531320e84ea Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Thu, 30 May 2019 09:54:57 +0300
Subject: [PATCH] Fix yocto build issues

- add tools-only option for building qtsafelayouttool for native and
  nativesdk targets
- remove uncessary RPATH
- remove host_build from the qtsafelayouttool, it depends on QtGui
  so it cannot be bootsrapped.
- use QT_PLUGIN_PATH for qtsafelayouttool calls. Builds are using qt.conf
  (from OE_QMAKE_QTCONF_PATH env in yocto builds) and from QT_HOST_BINS
  directory in the nativesdk, which has target paths for QT_INSTALL_PLUGINS
  that causes qtsafelayouttool to fail to find any plugins.

Change-Id: Ie042b7ef45c7a5e03b36008a2e2cbce1a9193139
---
 .qmake.conf                                 | 1 -
 src/src.pro                                 | 2 +-
 tools/featurespec/qtsaferenderer.prf        | 3 +++
 tools/qtsafelayouttool/qtsafelayouttool.pro | 1 -
 tools/tools.pro                             | 2 +-
 5 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/.qmake.conf b/.qmake.conf
index 72e3912..e5d116f 100644
--- a/.qmake.conf
+++ b/.qmake.conf
@@ -2,4 +2,3 @@ load(qt_build_config)
 CONFIG += warning_clean warnings_are_errors warn_on
 
 MODULE_VERSION = 1.1.0
-QMAKE_RPATHDIR += $$[QT_INSTALL_LIBS]
diff --git a/src/src.pro b/src/src.pro
index d5105c8..ee4912d 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -1,2 +1,2 @@
 TEMPLATE = subdirs
-SUBDIRS = qtsaferendererplugin messagesenderplugin saferenderer saferenderer/doc saferenderer/doc-unittests
+!tools-only:SUBDIRS = qtsaferendererplugin messagesenderplugin saferenderer saferenderer/doc saferenderer/doc-unittests
diff --git a/tools/featurespec/qtsaferenderer.prf b/tools/featurespec/qtsaferenderer.prf
index f133ed3..3a5c25e 100644
--- a/tools/featurespec/qtsaferenderer.prf
+++ b/tools/featurespec/qtsaferenderer.prf
@@ -1,3 +1,6 @@
+pluginpath.name = QT_PLUGIN_PATH
+pluginpath.value = $$[QT_HOST_LIBS/get]/plugins
+QT_TOOL_ENV += pluginpath
 qtPrepareTool(QMAKE_SAFELAYOUT, qtsafelayouttool, _DEP)
 qtPrepareTool(QMAKE_RCC, rcc, _DEP)
 
diff --git a/tools/qtsafelayouttool/qtsafelayouttool.pro b/tools/qtsafelayouttool/qtsafelayouttool.pro
index f3a62ea..d81e8c1 100644
--- a/tools/qtsafelayouttool/qtsafelayouttool.pro
+++ b/tools/qtsafelayouttool/qtsafelayouttool.pro
@@ -1,4 +1,3 @@
-option(host_build)
 QT += qml quick quick-private qml qml-private widgets
 
 CONFIG += c++11
diff --git a/tools/tools.pro b/tools/tools.pro
index 7b06ef9..333f26b 100644
--- a/tools/tools.pro
+++ b/tools/tools.pro
@@ -1,5 +1,5 @@
 TEMPLATE = subdirs
 SUBDIRS = featurespec
-!cross_compile: SUBDIRS += qtsafelayouttool
+tools-only:SUBDIRS += qtsafelayouttool
 
 
